### ISSUE TITLE
NullPointerException setImage(img) on windows platforms.

### ISSUE THREAD

**Issue by teocci** [dorkbox/SystemTray#91]
I created a trayicon menu using this library, the app worked nice some mins. However, suddenly I got a problem with the Java interface I was not able to select the windows of the app as shown in this 
[screenshot](https://i.imgur.com/oJ2PcZB.png).

The code that I run is this:

```
/**
* 트레이아이콘 이미지를 변경한다
*
* @param key 변경할 아이콘의 키 값
*/
public void setImage(String key) {
    if (_tray == null) return;
    try {
        Image img = _px > 16 ? CommonTheme.getImgBy24(key) : CommonTheme.getImg(key);

        if (img != null) {
            _tray.setImage(img); // << 1ST ERROR IS HERE
            _imgKey = key;
        }
    } catch (Exception e) {
        e.printStackTrace();
    }
}

public void startTrayIconChanger() {
    _isRunning = true;

    Thread cntGetter = new Thread(new Runnable() {
        @Override
        public void run() {
            while (_isRunning) {
                if (_main.getMailCnt() > 0) {
                    add(MAIL);
                } else {
                    remove(MAIL);
                }

                if (_main.getNoteCnt() > 0) {
                    add(NOTE);
                } else {
                    remove(NOTE);
                }

                if (_main.getApprCnt() > 0) {
                    add(APPR);
                } else {
                    remove(APPR);
                }

                for (int i : _indices) {
                    if (_tray == null) return;

                    Image img = _imgs[i];
                    if (img == null) return;

                    System.err.println("index: " +  i + img.toString());
                    _tray.setImage(img); // << 2ND ERROR IS HERE

                    try {
                        Thread.sleep(1000);
                    } catch (InterruptedException e) {
                        Debug.err(e);
                    }
                }

                if (_indices.size() > 0 && _trayMenu != null) {
                    _trayMenu.changeLang();
                }
            }
            // 종료된다면 로그아웃상태인것이므로 아이콘을 오프라인으로 표시한다
            setState(STrayIcon.BEFORE_LOGIN);
        }
    });

    cntGetter.setDaemon(true);
    cntGetter.start();
}
```
The error I got was this:

```
Exception in thread "Thread-13" java.lang.NullPointerException
	at dorkbox.util.jna.windows.HBITMAPWrap.createBitmap(HBITMAPWrap.java:73)
	at dorkbox.util.jna.windows.HBITMAPWrap.<init>(HBITMAPWrap.java:84)
	at dorkbox.systemTray.ui.swing._WindowsNativeTray.convertImage(_WindowsNativeTray.java:282)
	at dorkbox.systemTray.ui.swing._WindowsNativeTray.access$200(_WindowsNativeTray.java:57)
	at dorkbox.systemTray.ui.swing._WindowsNativeTray$1.setImage(_WindowsNativeTray.java:82)
	at dorkbox.systemTray.MenuItem.setImage_(MenuItem.java:147)
	at dorkbox.systemTray.SystemTray.setImage(SystemTray.java:1285)
	at kseek.msg.gui.tray.STrayIcon$1.run(STrayIcon.java:358)  << 1ST
	at java.lang.Thread.run(Thread.java:748)
java.lang.NullPointerException
	at dorkbox.util.jna.windows.HBITMAPWrap.createBitmap(HBITMAPWrap.java:73)
	at dorkbox.util.jna.windows.HBITMAPWrap.<init>(HBITMAPWrap.java:84)
	at dorkbox.systemTray.ui.swing._WindowsNativeTray.convertImage(_WindowsNativeTray.java:282)
	at dorkbox.systemTray.ui.swing._WindowsNativeTray.access$200(_WindowsNativeTray.java:57)
	at dorkbox.systemTray.ui.swing._WindowsNativeTray$1.setImage(_WindowsNativeTray.java:82)
	at dorkbox.systemTray.MenuItem.setImage_(MenuItem.java:147)
	at dorkbox.systemTray.SystemTray.setImage(SystemTray.java:1285)
	at kseek.msg.gui.tray.STrayIcon.setImage(STrayIcon.java:299) << 2ND
	at kseek.msg.gui.main.MainPn.setTrayIcon(MainPn.java:3984)
	at kseek.msg.gui.main.MainPn.apprArrived(MainPn.java:5647)
	at kseek.msglib.SegioMessenger.fireApprArrived(SegioMessenger.java:1014)
	at kseek.msglib.NotificationChannel.m_newappr(NotificationChannel.java:914)
	at kseek.msglib.NotificationChannel.processMessage(NotificationChannel.java:2168)
	at kseek.msglib.SegioChannel.readMessage(SegioChannel.java:391)
	at kseek.msglib.SegioChannel.run(SegioChannel.java:83)
```

**Comment by dorkbox** [comment]
Can you attach the images you are trying to set to the tray icon?

**Comment by dorkbox** [comment]
Also, updated source available, but I'm no longer using github to maintain my repos: https://git.dorkbox.com/dorkbox/SystemTray

**Comment by teocci** [comment]
This are the images: 
[logo.png](https://i.imgur.com/yZpMP6f.png)
[logo_offline.png](https://i.imgur.com/g5zddlY.png0)

Thank you for the update I will used the Version 3.15

**Comment by tresf** [comment]
> I'm no longer using github to maintain my repos: https://git.dorkbox.com/dorkbox/SystemTray

You have 116 project stars, 11 watchers and 19 forks.  Sorry to bring this off topic, but can I ask why?

**Comment by dorkbox** [comment]
>Sorry to bring this off topic, but can I ask why?

Two reasons. The data and information about me is not for sale, and unless absolutely necessary I don't let Microsoft stick their fingers in my life.

**Comment by Tristan971** [comment]
Very off-topic but hopefully the community comes up with a decentralized equivalent to it (a-la gitlab but all interconnected) and we can have the convenience of Github without our data being exploited. One can only wish...

**Comment by teocci** [comment]
I found [Gitee][1] this one is Chinese. And the developer is the "Chinese Software Developer Network" or "China Software Developer Network" aka CSDN, operated by Bailian Midami Digital Technology Co., Ltd., which is one of the biggest networks of software developers in China. But Chinese services are highly controlled by the government. So, it is the same. The question is "to whom you want to give your information."

 [1]: https://gitee.com/

**Comment by tresf** [comment]
> I don't let Microsoft stick their fingers in my life.

Microsoft has made great strides to help and work with the open source community.  Namely MSVC Core, VS Code, Open Sourcing the PowerShell framework, porting SQL Server to more platforms and open sourcing the Edge browser engine to name a few.  Microsoft us one of the largest users of the GitHub platform and it seems quite short-sighted to abandon the world's richest FOSS platform for speculative reasons.  Getting contributors to help `SystemTray` is already difficult due to the nonstandard build system.  Moving the source code away will ultimately lead to even less user and developer engagement.

That said, the decision seems to be already made so I'll (we'll) follow the project wherever it goes because it's the best library for what it does and I (we) truly appreciate the hard work that's gone into it.

**Comment by dorkbox** [comment]
> SystemTray is already difficult due to the nonstandard build system

It's finally all on Gradle now :)

Surprisingly, my build system was 1000% easier than Gradle -- everything was java, so the IDE did code completion and had full source documentation. Collecting files, compiling them, etc. I am a bit biased, mostly because Gradle is amazingly complicated if you want to actually do something other than `hello world`. For example, look [here](https://github.com/puniverse/quasar/blob/master/build.gradle) for a multi-project build that does something a tiny bit complicated.


 - Clone the [SystemTray](http://git.dorkbox.com/dorkbox/SystemTray.git) and [Utilities](https://git.dorkbox.com/dorkbox/Utilities) projects
 - open Utilities `build.gradle` in intellij
 - add SystemTray's `build.gradle` file to the Utilities project (+ button just to the left of the refresh gradle projects button)
 - done!

>  speculative reasons

I used to work for Microsoft as a contractor, and as long as the old-blood still runs through management, I don't trust them *at all*.  Hosting on my own git server was a difficult decision, because of how useful GitHub is, but ultimately I am not comfortable with Microsoft now owning GitHub. 

I'm leaving links up on GitHub (instead of just deleting the projects), and all the search engines now crawl my server, so I'm not worried people won't be able to find and/or contribute. It's hosted via Gitea, and looks/feels almost identical to github.

**Comment by dorkbox** [comment]
This should now be fixed in Version 3.17
